src/Lecture2.hs

```diff
@@ -198,7 +266,10 @@ verify that.
 [1,2,3,4,7]
 -}
 merge :: [Int] -> [Int] -> [Int]
-merge = error "TODO"
+merge [] ys = ys
+merge xs [] = xs
+merge (x:xs) (y:ys) = if x <= y then [x] ++ merge (xs) (y:ys)
+                      else [y] ++ merge (x:xs) (ys)
```

Comment:
Absolutely correct solution! 💎 
You can make it more performant by checking if both heads of the given lists are equal. In that case, you're able to reduce the size of each list by 1 on each recursive call instead of reducing the size of only a single list ✂️

Also, you can always rewrite `[x] ++ l` as `x : l` and it's usually more efficient

Reply:
👍🏼 Thanks for the suggestion.